src/SourceGenerators/Uno.UI.SourceGenerators/ImplementedRoutedEvents/ImplementedRoutedEventsGenerator.cs

```diff
@@ -0,0 +1,183 @@
+				_events.Add("OnDragEnter", dragEventArgs);
+				_events.Add("OnDragOver", dragEventArgs);
+				_events.Add("OnDragLeave", dragEventArgs);
+				_events.Add("OnDrop", dragEventArgs);
+
+				var keyRoutedEventArgs = GetRequiredTypeByMetadataName("Windows.UI.Xaml.Input.KeyRoutedEventArgs", context.Compilation);
+				_events.Add("OnKeyDown", keyRoutedEventArgs);
+				_events.Add("OnKeyUp", keyRoutedEventArgs);
+
+				var routedEventArgs = GetRequiredTypeByMetadataName("Windows.UI.Xaml.RoutedEventArgs", context.Compilation);
+				_events.Add("OnLostFocus", routedEventArgs);
+				_events.Add("OnGotFocus", routedEventArgs);
+			}
+
+			private static INamedTypeSymbol GetRequiredTypeByMetadataName(string fullyQualifiedMetadataName, Compilation compilation)
+			{
+				var type = compilation.GetTypeByMetadataName(fullyQualifiedMetadataName);
+				return type ?? throw new InvalidOperationException($"The type '{fullyQualifiedMetadataName}' is not found.");
+			}
+
+			public override void VisitAssembly(IAssemblySymbol symbol) => base.VisitAssembly(symbol);
+			public override void VisitModule(IModuleSymbol symbol) => VisitNamespace(symbol.GlobalNamespace);
+
+			public override void VisitNamedType(INamedTypeSymbol symbol)
+			{
+				foreach (var type in symbol.GetTypeMembers())
+				{
+					VisitNamedType(type);
+				}
+
+				ProcessType(symbol);
+			}
+
+			public override void VisitNamespace(INamespaceSymbol symbol)
+			{
+				foreach (var n in symbol.GetNamespaceMembers())
+				{
+					VisitNamespace(n);
+				}
+
+				foreach (var t in symbol.GetTypeMembers())
+				{
+					VisitNamedType(t);
+				}
+			}
+
+			private void ProcessType(INamedTypeSymbol type)
+			{
+				// Control is defining the virtual property, we cannot generate an override for it.
+				// Use of _seenTypes to prevent processing the same type twice, which causes warnings like:
+				// warning CS2002: Source file 'src\Uno.UI\obj\Debug\monoandroid11.0\g\ImplementedRoutedEventsGenerator\TwoPaneView_ImplementedRoutedEvents_g_cs.g.cs' specified multiple times
+				if (!type.IsAbstract && type.Is(_controlSymbol) && !type.Equals(_controlSymbol, SymbolEqualityComparer.Default) && _seenTypes.Add(type))
+				{
+					var list = new List<string>();
+					list.Add("global::Uno.UI.Xaml.RoutedEventFlag.None");
+
+					foreach (var @event in _events)
+					{
+						if (!@event.Key.StartsWith("On", StringComparison.Ordinal))
+						{
+							throw new InvalidOperationException("Expected event to start with 'On'");
+						}
+
+						if (IsEventOverrideImplemented(type, @event.Key, @event.Value))
+						{
+							list.Add($"global::Uno.UI.Xaml.RoutedEventFlag.{@event.Key.Substring(2)}");
+						}
+					}
+
+					GenerateCode(type, list);
+				}
+			}
+
+			private bool IsEventOverrideImplemented(INamedTypeSymbol type, string name, INamedTypeSymbol arg)
+			{
+				var method = type.GetMembers(name).SingleOrDefault(
```

Comment:
⚠️ ⚠️ Likely I'll need to retrieve members from the inheritance hierarchy until reaching `Control` (not including it)